src/index/IndexImpl.Text.cpp

```diff
@@ -803,7 +803,31 @@ Index::WordEntityPostings IndexImpl::readWordCl(
 }
 
 // _____________________________________________________________________________
-Index::WordEntityPostings IndexImpl::readWordEntityCl(
+IdTable IndexImpl::readWordCl(
+    const TextBlockMetaData& tbmd,
+    const ad_utility::AllocatorWithLimit<Id>& allocator) const {
+  IdTable idTable{allocator};
+  idTable.setNumColumns(2);
+  vector<TextRecordIndex> cids = readGapComprList<TextRecordIndex>(
+      tbmd._cl._nofElements, tbmd._cl._startContextlist,
+      static_cast<size_t>(tbmd._cl._startWordlist - tbmd._cl._startContextlist),
+      &TextRecordIndex::make);
+  idTable.resize(cids.size());
+  std::ranges::transform(cids, idTable.getColumn(0).begin(),
+                         &Id::makeFromTextRecordIndex);
+  std::ranges::transform(
+      readFreqComprList<WordIndex>(
+          tbmd._cl._nofElements, tbmd._cl._startWordlist,
+          static_cast<size_t>(tbmd._cl._startScorelist -
+                              tbmd._cl._startWordlist)),
+      idTable.getColumn(1).begin(), [](WordIndex id) {
+        return Id::makeFromWordVocabIndex(WordVocabIndex::make(id));
+      });
+  return idTable;
```

Comment:
You already pointed out the ideas to make this more efficient, but possibly not in this PR.